### PR TITLE
fix：sofa-ark-plugin-maven-plugin remove unnecessary shade lib

### PR DESCRIPTION
**Pull request by yuandongjian**
fix [issue-935](https://github.com/sofastack/sofa-ark/issues/935)

## Summary by CodeRabbit

- **Bug Fixes**
  - Corrected the artifact handling in the archiver to avoid adding conflicting project artifacts.
  - Updated artifact declarations and settings to ensure proper handling and cleanup in test cases.

- **Refactor**
  - Simplified artifact conditions and dependency checks for more efficient processing in the build plugin.

- **Tests**
  - Adjusted test cases to reflect changes in artifact handling and ensure accurate validation of resources.

### CODE REVIEW

**Comment by lvjing2** [comment]
Hi, Please don't forget to assign the CLA

**Comment by yuandongjian** [comment]
> Hi, Please don't forget to assign the CLA

已经签署完成

File: sofa-ark-parent/support/ark-plugin-maven-plugin/src/main/java/com/alipay/sofa/ark/plugin/mojo/ArkPluginMojo.java [hunk 1]
```diff
@@ -281,9 +281,8 @@ public boolean isShadeJar(Artifact artifact) {
      */
     protected void addArkPluginArtifact(Archiver archiver, Set<Artifact> dependencies,
                                         Set<Artifact> conflicts) {
-        addArtifact(archiver, project.getArtifact(), conflicts.contains(project.getArtifact()));
         for (Artifact artifact : dependencies) {
-            if (Repackager.isZip(artifact.getFile())) {
+            if (Repackager.isZip(artifact.getFile()) && !isShadeJar(artifact)) {
```
[the remark this answers is not in the document]
Comment (yuandongjian) [reply sofa-ark-parent/support/ark-plugin-maven-plugin/src/main/java/com/alipay/sofa/ark/plugin/mojo/ArkPluginMojo.java]: PR里增加了isShadeJar的判断，目的也是为了去掉在lib/目录里已经shade过的jar。
不过在addArtifact里，也会判断isShadeJar，所以这行改动确实没必要，已经去掉了。

**Review by lvjing2 — APPROVED** [review]
LGTM

**Pull request merged**